### ISSUE TITLE
Read data by URL

### ISSUE THREAD

**Issue by zverok** [SciRuby/daru#289]
Another one stolen from our competitors (as #127). They can do this:

```r
df = pandas.read_csv('https://download.data.world/sql_query_result_download/producthunt/product-hunt-research?filename=product-hunt-research-QueryResult&mimetype=text%2Fcsv&query=--%20NOTE%3A%20from%20a%20sample%20of%205000%20records%20in%20the%20full%20dataset%0ASELECT%20name%2C%20time_of_day%2C%20PostsForExploration.date%0AFROM%20PostsForExploration%20&auth=<USER-FILE-SPECIFIC-TOKEN>')
```

With Ruby's `open-uri`, it would not be too hard to reproduce, and, again, not that useful -- but "cool looking" on demos.

**Comment by v0dro** [comment]
Works.

**Comment by Varunram** [comment]
@v0dro Do we add this as an extension?

**Comment by v0dro** [comment]
Extension as in?

**Comment by anshuman23** [comment]
Submitted the PR for this issue. I will also work on #127 

Please let me know if I there is anything that I can improve on in the PR.

**Comment by athityakumar** [comment]
Fixed with PR #334 - I think this issue can be closed now. :smile:

**Comment by zverok** [comment]
:+1:
